### DISCUSSION TITLE
objects appear to be null when debugging

### DISCUSSION THREAD

**Discussion by edgedj** [dotnet/maui#7647] [Q&A]
has anyone noticed that objects are null when debugging an external library? I have a Maui application I'm building and when I step out to a net6.0 library which has targets including maccatalyst, as I step through the code many objects are null when in fact they aren't.

Take this example:

```
                SpreadsheetDocument spreadSheet = SpreadsheetDocument.Open(templateFileLocation, false);
                WorkbookPart wb = spreadSheet.WorkbookPart;
                var allParts = wb.Parts.ToList();
                Console.WriteLine("allParts size=" + allParts.Count().ToString());
```
as I step through, allParts is null however in the console I can see that it is in fact a list with 14 items.

**Comment by campersau** [comment]
There are some issues about that:
- https://github.com/dotnet/maui/issues/7555
- https://github.com/dotnet/maui/issues/7009

**Comment by beeradmoore** [comment]
Can you defined allParts as a class variable (RIP var keyword) and not a local one and see if the problem goes away?
